packages/flutter/test/material/chip_test.dart

```diff
@@ -1017,417 +987,750 @@ void main() {
-    // Creates a chip with a delete button.
-    final UniqueKey labelKey = UniqueKey();
-    final UniqueKey deleteButtonKey = UniqueKey();
+  group('Ink features', () {
+    group('with InkSplash splash factory', () {
+      testWidgets('Chip creates centered, unique splash when label is tapped', (WidgetTester tester) async {
+        final UniqueKey labelKey = UniqueKey();
+        final UniqueKey deleteButtonKey = UniqueKey();
+
+        await tester.pumpWidget(
+          _chipWithOptionalDeleteButton(
+            splashFactory: InkSplash.splashFactory,
+            labelKey: labelKey,
+            deleteButtonKey: deleteButtonKey,
+            deletable: true,
+          ),
+        );
 
-    await tester.pumpWidget(
-      _chipWithOptionalDeleteButton(
-        labelKey: labelKey,
-        deleteButtonKey: deleteButtonKey,
-        deletable: true,
-      ),
-    );
+        final RenderBox box = getMaterialBox(tester);
 
-    final RenderBox box = getMaterialBox(tester);
+        // Taps at a location close to the center of the label.
+        final Offset centerOfLabel = tester.getCenter(find.byKey(labelKey));
+        final Offset tapLocationOfLabel = centerOfLabel + const Offset(-10, -10);
+        final TestGesture gesture = await tester.startGesture(tapLocationOfLabel);
+        await tester.pump();
 
-    // Taps at a location close to the center of the label.
-    final Offset centerOfLabel = tester.getCenter(find.byKey(labelKey));
-    final Offset tapLocationOfLabel = centerOfLabel + const Offset(-10, -10);
-    final TestGesture gesture = await tester.startGesture(tapLocationOfLabel);
-    await tester.pump();
+        await tester.pump(const Duration(milliseconds: 100));
 
-    // Waits for 100 ms.
-    await tester.pump(const Duration(milliseconds: 100));
+        expect(find.byType(InkWell), findsOneWidget);
+        expect(find.byType(InkResponse), findsNothing);
 
-    // There should be exactly one ink-creating widget.
-    expect(find.byType(InkWell), findsOneWidget);
-    expect(find.byType(InkResponse), findsNothing);
+        expect(box, paintsRipple(center: const Offset(163, 6), radius: 21)); // Splash is centered and growing
+        expect(box, paintsRipple(center: const Offset(163, 6), radius: 21, unique: true));
 
-    // There should be one unique, centered ink ripple.
-    expect(box, ripplePattern(const Offset(163.0, 6.0), 20.9));
-    expect(box, uniqueRipplePattern(const Offset(163.0, 6.0), 20.9));
+        expect(findTooltipContainer('Delete'), findsNothing);
 
-    // There should be no tooltip.
-    expect(findTooltipContainer('Delete'), findsNothing);
+        await tester.pump(const Duration(milliseconds: 100));
 
-    // Waits for 100 ms again.
-    await tester.pump(const Duration(milliseconds: 100));
+        expect(box, paintsRipple(center: const Offset(163, 6), radius: 42));
+        expect(box, paintsRipple(center: const Offset(163, 6), radius: 42, unique: true));
```

Comment:
I'm wondering if we could remove this double-check here (and in other similar places) and leave only the unique one.
Am I missing some reason it's checked twice with unique and non-unique?

Reply:
If `unique` defaults to `false`, proper coverage would test both cases.

Reply:
Good question—I'd be happy to remove it if you'd prefer. Largely, I just saw that the previous test was testing for a ripple, and then for a unique ripple, and I thought it would be imprudent to arbitrarily take away expectations on a test that I'm refactoring, when I'm not actually changing the expected behavior related to the uniqueness?